### PR TITLE
[FIX] Autodetect encode + specify encode mode for autodetect

### PR DESCRIPTION
**Pull request by Imipenem**
**PR Checklist**

-   [x] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes**

- fixed an issue that caused ehrapy to crash when there are no columns that needs to be encoded (all numerical columns, #290)

- using autodetect while encoding now allows to pass a specific encoding mode used for the encoding (default is label encoding; works only for single column encoding modes like count, one-hot and label encoding #287 )

- added tests for the fix and feature

### CODE REVIEW

**Comment by Imipenem** [comment]
@Zethson: Unfortunately the `type: ignore` are necessary since, we are indeed checking for type before to ensure we work with the correct data type but `mypy` does not recognize this.

File: tests/preprocessing/test_encode.py [hunk 1]
```diff
@@ -42,11 +42,36 @@ def test_autodetect_encode(self):
         }
         assert id(encoded_ann_data.X) != id(encoded_ann_data.layers["original"])
 
+    def test_autodetect_num_only(self, capfd):
+        adata = read(dataset_path=f"{_TEST_PATH}/dataset2.csv")
+        encoded_ann_data = encode(adata, autodetect=True)
+        out, err = capfd.readouterr()
```
Comment (Zethson) [inline tests/preprocessing/test_encode.py:48]: I'd prefer a simple assert whether the error was thrown. No need to check the message. If we ever were to change the error message we would also have to change it here. Annoying.

**Review by Zethson — CHANGES_REQUESTED** [review]
Minor changes again, thanks!

File: tests/preprocessing/test_encode.py (hunk 1 above)
Comment (Imipenem) [reply tests/preprocessing/test_encode.py:48]: Would have done this normally, yes. The thing is, we do not throw an error here (we just print a warning).

Comment (Zethson) [reply tests/preprocessing/test_encode.py:48]: Ah okay.I get it. The message is ugly AF though :)

Should be more like:

"No columns to be encoded were detected" :)

**Pull request merged**